nemo_curator/modules/exact_dedup.py

```diff
@@ -166,10 +173,38 @@ def __call__(self, dataset: DocumentDataset) -> Union[DocumentDataset, str]:
         self._logger.info(
             f"Time taken for Exact Dedup Computation = {time.time() - t0}s and output written at {write_path}"
         )
-        if is_cudf_type(result):
-            import dask_cudf
+        backend = "cudf" if is_cudf_type(result) else "pandas"
+        return DocumentDataset.read_parquet(
+            write_path,
+            backend=backend,
+            blocksize="1024MiB",
```

Comment:
Somewhat related: `blocksize=None` is a valid approach when wanting to read in 1 row group per partition or 1 fpp in the dask world, while DocDataset doesn't work well when both this and app is set to None.